File: hospital.client/src/types/VitalSignResponse.ts
export interface VitalSignResponse {
  id: number;
  appointmentId: number;
  nurseId: number;
  bloodPressureSystolic: number;
  bloodPressureDiastolic: number;
  temperature: number;
  weight: number;
  height: number;
  heartRate: number;
  isEmergency: boolean;
  clinicalAlerts?: string | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;
}

export interface VitalSignRequest {
  id?: number | null;
  appointmentId?: number | null;
  nurseId?: number | null;
  bloodPressureSystolic?: number | null;
  bloodPressureDiastolic?: number | null;
  temperature?: number | null;
  weight?: number | null;
  height?: number | null;
  heartRate?: number | null;
  isEmergency?: boolean | null;
  clinicalAlerts?: string | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}

File: Hospital.Server/Entities/Response/VitalSignResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class VitalSignResponse
    {
        public long Id { get; set; }
        public long AppointmentId { get; set; }
        public long NurseId { get; set; }
        public int BloodPressureSystolic { get; set; }
        public int BloodPressureDiastolic { get; set; }
        public decimal Temperature { get; set; }
        public decimal Weight { get; set; }
        public decimal Height { get; set; }
        public int HeartRate { get; set; }
        public bool IsEmergency { get; set; }
        public string? ClinicalAlerts { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
    }
}
